packit/cli/builds/obs_build.py

```diff
@@ -0,0 +1,309 @@
+)
+@click.option(
+    "--project",
+    help="Project name to build in. It will be created if does not exist."
+    " It defaults to home:$owner:packit:$pkg",
+)
+@click.option(
+    "--targets",
+    help="Comma separated list of chroots to build in. (defaults to 'fedora-rawhide-x86_64')",
+    default="fedora-rawhide-x86_64",
+)
+@click.option(
+    "--description",
+    help="Description of the project to build in.",
+    default=None,
+)
+@click.option(
+    "--upstream-ref",
+    default=None,
+    help="Git ref of the last upstream commit in the current branch "
+    "from which packit should generate patches "
+    "(this option implies the repository is source-git).",
+)
+@click.option("--wait/--no-wait", default=True, help="Wait for the build to finish")
+@click.option(
+    PACKAGE_SHORT_OPTION,
+    PACKAGE_LONG_OPTION,
+    multiple=True,
+    help=PACKAGE_OPTION_HELP.format(action="build"),
+)
+@click.argument("path_or_url", type=LocalProjectParameter(), default=os.path.curdir)
+@pass_config
+@cover_packit_exception
+@iterate_packages
+def obs(
+    config: Config,
+    owner: Optional[str],
+    project: str,
+    targets: str,
+    description: Optional[str],
+    upstream_ref: Optional[str],
+    wait: bool,
+    package_config: PackageConfig,
+    path_or_url,
+) -> None:
+    api = get_packit_api(
+        config=config,
+        package_config=package_config,
+        local_project=path_or_url,
+    )
+
+    conf.get_config()
+    owner = owner or conf.config["api_host_options"][_API_URL]["user"]
+    prj_name = project or f"home:{owner}:packit"
+
+    targets_list = targets.split(",")
+    for target in targets_list:
+        if target in DEPRECATED_TARGET_MAP:
+            logger.warning(
+                f"Target '{target}' is deprecated. "
+                f"Please use '{DEPRECATED_TARGET_MAP[target]}' instead.",
+            )
+
+    prj_meta = targets_to_project_meta(
+        targets=targets_list,
+        owner=owner,
+        prj_name=prj_name,
+        description=description,
+    )
+
+    url = core.makeurl(_API_URL, ["source", prj_name, "_meta"])
+    mf = core.metafile(url, ET.tostring(prj_meta))
+    mf.sync()
+
+    pkg_names = list(package_config.packages.keys())
+
+    if len(pkg_names) != 1:
+        raise ValueError("Cannot handle multiple packages in package_config")
+
+    create_package(prj_name, (pkg_name := pkg_names[0]))
+
+    with TemporaryDirectory() as tmp_dir:
+        core.Project.init_project(_API_URL, (prj_dir := Path(tmp_dir)), prj_name)
+
+        (pkg_dir := (prj_dir / pkg_name)).mkdir()
+        core.checkout_package(
+            _API_URL,
+            prj_name,
+            pkg_name,
+            prj_dir=prj_dir,
+            pathname=pkg_dir,
+        )
+
+        pkg = core.Package(pkg_dir)
+
+        for fname in filter(os.path.isfile, os.listdir(pkg_dir)):
+            pkg.delete_file(fname)
+
+        srpm = api.create_srpm(upstream_ref=upstream_ref, release_suffix="0")
```

Comment:
why is the `release_suffix` being set to 0 here?

Reply:
OBS sets the release field itself as it by default rebuilds everything automatically and thus hardcoding the release field makes no sense in the context of OBS